joystreamjs/content/src/channelPayouts.ts

```diff
@@ -0,0 +1,283 @@
+import { ChannelPayoutsMetadata, IChannelPayoutsMetadata } from '@joystream/metadata-protobuf'
+import { createType } from '@joystream/types'
+import { asValidatedMetadata, readBytesFromFile, ReadFileContext } from '@joystreamjs/utils'
+import { ChannelPayoutProof } from '@joystreamjs/utils/typings/ChannelPayoutsPayload.schema'
+import { ChannelPayoutsVector } from '@joystreamjs/utils/typings/ChannelPayoutsVector.schema'
+import { u8aToHex } from '@polkadot/util'
+import { blake2AsU8a } from '@polkadot/util-crypto'
+import BN from 'bn.js'
+import Long from 'long'
```

Comment:
I was wondering, why do we use both `Long` and `BN`? I thought they are used for the same thing?

Reply:
They aren't really the same thing, As you know, JS numbers are 53 bits long (not big enough), `Long` provides some more bits for integer representation, it represents numbers 64 bits 2's complements integer, and the protobuf JS library decodes `uint64` fields to `Long` types by default. Lastly, `BN` is used to represent arbitrary large numbers, and it works really well with the PolkadotJS as `u64` and `u128` in javascript exist in BN representation